Source/SwiftLintFramework/Extensions/NSRegularExpression+SwiftLint.swift

```diff
@@ -8,13 +8,21 @@
 
 import Foundation
 
+private var regexCache = [String: NSRegularExpression]()
+
 public extension NSRegularExpression {
-    public convenience init(pattern: String) throws {
-        try self.init(pattern: pattern,
+    public static func cached(pattern pattern: String) throws -> NSRegularExpression {
```

Comment:
Why not keep the old API (`NSRegularExpression.init(pattern:)`)? We should probably start being more conservative about API breaking changes if we want to prepare for 1.0 :smile:.